packages/matter.js/src/log/Logger.ts

```diff
@@ -184,33 +184,20 @@ const defaultValueFormatter = (value: string) => value;
  * log output.  See Logger.dict() for maximal convenience.
  */
 export class DiagnosticDictionary {
-    private entries = new Map<string, any>();
-
     /**
      * Create a new dictionary with optional entry values.
      *
      * @param entries the entries as [ "KEY", value ] tuples
      */
-    constructor(entries: { [KEY: string]: any } = {}) {
-        // Use getOwnPropertyNames because it follows insertion order as of
-        // ES6
+    constructor(private readonly entries: { [KEY: string]: any } = {}) {
+        // Use getOwnPropertyNames because it follows insertion order as of ES6
         for (const KEY of Object.getOwnPropertyNames(entries)) {
-            if (entries[KEY] !== undefined) {
-                this.set(KEY, entries[KEY]);
+            if (entries[KEY] === undefined) {
+                delete entries[KEY];
```

Comment:
This is destructive to input object, maybe could just omit `undefined` in serializion?

Reply:
Good idea. Will do